jax/_src/lax/lax.py

```diff
@@ -1338,11 +1339,12 @@ def full_like(x: Array, fill_value: ArrayLike, dtype: Optional[DTypeLike] = None
   # probably in the form of a primitive like `val = match_sharding_p.bind(x, val)`
   # (so it works in staged-out code as well as 'eager' code). Related to
   # equi-sharding.
-  if (config.jax_array and hasattr(x, 'sharding') and
-      not dispatch.is_single_device_sharding(x.sharding) and
-      not isinstance(x.sharding, sharding.PmapSharding)):
-    return array.make_array_from_callback(
-        fill_shape, x.sharding, lambda idx: val[idx])  # type: ignore[arg-type]
+  if config.jax_array and shape is None and hasattr(x, 'sharding'):
+    sharding = x.sharding  # type: ignore[union-attr]
```

Comment:
I think you can call this `xs = x.sharding` and still import sharding. But I am fine either way. 

Its just if I remove PmapSharding from array, then this will break.

Reply:
yeah, changed it to import `PmapSharding` from `sharding`